test/remote/account_signin_verification_tests.js

```diff
@@ -178,172 +178,6 @@ TestServer.start(config)
     )
 
     test(
-      'Unverified account becomes verified from sign-in verification',
-      function (t) {
-        var email = server.uniqueEmail()
-        var password = 'allyourbasearebelongtous'
-        var client = null
-        var uid
-        var code
-        return Client.create(config.publicUrl, email, password)
-          .then(
-            function (x) {
-              client = x
-              t.ok(client.authAt, 'authAt was set')
-            }
-          )
-          .then(
-            function () {
-              return client.emailStatus()
-            }
-          )
-          .then(
-            function (status) {
-              t.equal(status.verified, false, 'new account is unverified')
-              t.equal(client.emailVerified, false, 'new account email unverified')
-            }
-          )
-          .then(
-            function () {
-              return server.mailbox.waitForEmail(email)
-            }
-          )
-          .then(
-            function (emailData) {
-              t.equal(emailData.subject, 'Verify your Firefox Account')
-            }
-          )
-          .then(
-            function () {
-              return client.login({keys:true})
-            }
-          )
-          .then(
-            function () {
-              return server.mailbox.waitForEmail(email)
-            }
-          )
-          .then(
-            function (emailData) {
-              uid = emailData.headers['x-uid']
-              code = emailData.headers['x-verify-code']
-              t.equal(emailData.subject, 'Confirm new sign-in to Firefox')
-              t.ok(uid, 'sent uid')
-              t.ok(code, 'sent verify code')
-            }
-          )
-          .then(
-            function () {
-              return client.emailStatus()
-            }
-          )
-          .then(
-            function (status) {
-              t.equal(status.verified, false, 'account is not verified, unverified sign-in')
-            }
-          )
-          .then(
-            function () {
-              return client.verifyEmail(code)
-            }
-          )
-          .then(
-            function () {
-              return client.emailStatus()
-            }
-          )
-          .then(
-            function (status) {
-              t.equal(status.verified, true, 'account is verified by confirming email')
-            }
-          )
-      }
-    )
-
-    test(
-      'Account verification links still work after session verification',
```

Comment:
I think this test is worth keeping, changed to use two separate "verify your account" emails rather than a "verify" and a "confirm".  IIUC all the underlying logic should be the same, e.g. verifying via the link in the second email shouldn't break verifying by the link in the first email.  We can also check that the two verification codes are in fact different.